internal/dot/graph.go

```diff
@@ -252,6 +270,50 @@ func (dg *Graph) addToGroup(r *Result, id CtorID) {
 	group.Results = append(group.Results, r)
 }
 
+// Prune removes elements from the graph that do not have failed results.
+// Removing elements that do not have failing results makes the graph easier to debug, since non-failing
+// nodes and edges can clutter the graph and don't help the user debug.
+func (dg *Graph) Prune() {
+	failedResults := make(map[*Result]bool)
+	for _, v := range dg.Failed.RootCauses {
+		failedResults[v] = true
+	}
+	for _, v := range dg.Failed.TransitiveFailures {
+		failedResults[v] = true
```

Comment:
nit: For sets, we generally use `struct{}` instead of `bool`. It doesn't
matter much here but `struct{}` is zero-cost in terms of space, IIRC.

```
make(map[foo]struct[})
```

Reply:
Resolved